server/src/main/scala/com/ossuminc/riddl/lsp/server/RiddlLSPServer.scala

```diff
@@ -1,8 +1,34 @@
 package com.ossuminc.riddl.lsp.server
 
-object RiddlLSPServer {
+import org.eclipse.lsp4j.{CompletionOptions, InitializeParams, InitializeResult, ServerCapabilities, TextDocumentSyncKind}
+import org.eclipse.lsp4j.services.{LanguageServer, TextDocumentService, WorkspaceService}
 
-  final def main(args: Array[String]): Unit = {
-    System.exit(0)
+import java.util.concurrent.CompletableFuture
+
+object RiddlLSPServer extends LanguageServer {
+  var errorCode = 0
```

Comment:
I'm not a fan of this use of a var to capture an errorCode. I presume you have a good reason for this var? Usually, when I must write a var, I also put a comment indicating why it is required.

Reply:
This resulted from translating Java code into the skeleton. Ideally we would have probably an Actor system for state management? Maybe something other than an Actor system?